tests/e2e/nvidia-container-toolkit_test.go

```diff
@@ -257,4 +257,32 @@ var _ = Describe("docker", Ordered, ContinueOnFailure, func() {
 			Expect(libs).To(ContainElements([]string{"libcuda.so", "libcuda.so.1"}))
 		})
 	})
+
+	When("A container tries to create firmware files on the host", Ordered, func() {
+		BeforeAll(func(ctx context.Context) {
+			_, _, err := runner.Run("docker pull ubuntu")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, err = runner.Run(`docker build -t firmware-test --build-arg RM_VERSION="$(basename $(ls -d /lib/firmware/nvidia/*.*))" --build-arg CURRENT_DIR="$(pwd)" - <<EOF
+FROM ubuntu
+RUN mkdir -p /lib/firmware/nvidia/
+ARG RM_VERSION
+ARG CURRENT_DIR
+RUN ln -s /../../../../../../../../\$CURRENT_DIR /lib/firmware/nvidia/\$RM_VERSION
+EOF`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should not create empty firmware files on the host when using CDI", func(ctx context.Context) {
+			output, _, err := runner.Run("docker run --rm --runtime=nvidia --gpus=all firmware-test")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(BeEmpty())
+		})
+
+		It("should fail when using the pre-start hook", func(ctx context.Context) {
```

Comment:
De we refer to this as "legacy" mode in other tests?

Reply:
test description edited